intake/container/base.py

```diff
@@ -1,21 +1,81 @@
-class BaseContainer(object):
+import msgpack
+import requests
+from intake.source.base import DataSource, Schema
+from . import serializer
+from intake import __version__
 
-    @staticmethod
-    def merge(parts):
-        raise NotImplementedError
 
-    @staticmethod
-    def to_dask(parts, dtype):
-        raise NotImplementedError
+class RemoteSource(DataSource):
+    version = __version__
 
-    @staticmethod
-    def encode(obj):
-        raise NotImplementedError
+    def __init__(self, url, headers, name, parameters, metadata=None, **kwargs):
+        super(RemoteSource, self).__init__(self)
+        self.url = url
+        self.name = name
+        self.parameters = parameters
+        self.headers = headers
+        self._source_id = None
+        self.metadata = metadata or {}
+        self._get_source_id()
 
-    @staticmethod
-    def decode(bytestr):
-        raise NotImplementedError
+    def _get_source_id(self):
+        if self._source_id is None:
+            payload = dict(action='open', name=self.name,
+                           parameters=self.parameters)
+            req = requests.post(self.url, data=msgpack.packb(
+                payload, use_bin_type=True), **self.headers)
+            if req.status_code == 200:
```

Comment:
Do we need to raise an exception here?  We do that above.

Reply:
done